File: airbyte-cdk/java/airbyte-cdk/dependencies/src/main/java/io/airbyte/commons/functional/CheckedSupplier.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.commons.functional;

import org.apache.commons.lang3.function.FailableSupplier;

@FunctionalInterface
public interface CheckedSupplier<T, E extends Throwable> extends FailableSupplier<T, E> {

}

File: airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/commons/functional/CheckedSupplier.kt
/*
 * Copyright (c) 2026 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.commons.functional

import org.apache.commons.lang3.function.FailableSupplier

fun interface CheckedSupplier<T, E : Throwable?> : FailableSupplier<T, E>
